### PR TITLE
Added release checklist

### PR DESCRIPTION
**Pull request by InnayTool**
### Component/Part
development documentation

### Description
This PR adds a release checklist which should be followed. Do you have more suggestions?

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/master/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: docs/dev/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+- [ ] Bump the version
+- [ ] Make sure `yarn.lock` is up to date
+- [ ] Make sure translations are up to date
+- [ ] Write release notes
+  - [ ] General description
+  - [ ] Things requiring special action beside updating the software
+  - [ ] New features
+  - [ ] Bug fixes
+  - [ ] Add all contributors
+- [ ] Run the tests described in the release test document (internal)
```
Comment (ErikMichelson) [inline docs/dev/release-checklist.md:13]: Do we just want to include these tests here?

Comment (InnayTool) [reply docs/dev/release-checklist.md:13]: Yes, I think it's a good idea to include a more specific description of what to test.

**Comment by mrdrogdrog** [comment]
After https://github.com/hedgedoc/hedgedoc.github.io/issues/44 is merged we should add "Build and deploy homepage" to the list

image

File: docs/dev/release-checklist.md [hunk 2]
```diff
@@ -13,3 +13,4 @@
 - [ ] Run the tests described in the release test document (internal)
 - [ ] Upload tar ball to github and add a release tag
 - [ ] Update container images
+- [ ] Update Website by clicking [workflow button](https://github.com/hedgedoc/hedgedoc.github.io/actions) in hedgedoc/hedgedoc.github.io
```
Comment (mrdrogdrog) [inline docs/dev/release-checklist.md:16]:
```
Update website by running the [deploy workflow](https://github.com/hedgedoc/hedgedoc.github.io/actions?query=workflow%3A%22Deploy+to+github+actions+branch%22) in hedgedoc/hedgedoc.github.io
```

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]

**Comment by ErikMichelson** [comment]
To add:
 - Update docker-compose in the docs

**Pull request closed without merging**